pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -586,6 +586,10 @@ func TestClearStaleGroups(t *testing.T) {
 
 			mctrl.clearStaleGroups()
 
+			// Add an invalid event into the queue as a mark to ensure all expected events are processed before
+			// the case is completed.
+			mctrl.queue.Add("unexistgroup")
```

Comment:
I am actually not sure the fix is correct. We may still miss be missing events that have not been added to the queue yet?

Reply:
The event is added into the queue before we wait the queue to be empty, it means when we start to check the length of the queue (https://github.com/antrea-io/antrea/pull/7183/files/b0e3c82dbc2b2e17a56a721796d9db85b4b07ae1#diff-04e5b5e2a54a1d059c1ba078b13ccb0a2469a051d33fc6c6f203f2d052da8595R593), it has at least one item (the "unexistgroup"). So if the condition is satisfied, meaning the "unexistinggroup" is got from the queue, and the previous events have been processed.

Besides, since I have modified the `igmpMaxResponseTime` as -1, it means the Multicast group is actually added into the queue at once when `clearStaleGroups` calls `queue.AddAfter`. This ensures that the expected events in the queue to be added before "unexistinggroup"

Reply:
Maybe I missed one scenario, the `clearStaleGroups` also adds IGMP event to the groupEventCh, and `eventHandler` (the consumer of the groupEventCh) will adds the item into the queue. Let me update the fix as below,

1. Add invalid IGMP leave event into the groupEventCh
2. Wait until the `groupEventCh` is empty, which means the generated IGMP events are processed 
3. Add invalid group item into the queue
4. Wait until the queue is empty.

What do you think? @antoninbas

Reply:
I wonder if we could avoid steps 3 and 4 by using `ShutDownWithDrain` instead of `ShutDown`?

Reply:
> I wonder if we could avoid steps 3 and 4 by using `ShutDownWithDrain` instead of `ShutDown`?

Sure, I will update.